irctest/server_tests/chmodes/nooutside.py

```diff
@@ -0,0 +1,38 @@
+"""
+Channel "no outside" mode (`RFC 1459
+<https://datatracker.ietf.org/doc/html/rfc1459#section-4.2.3.1>`__,
+`Modern <https://modern.ircdocs.horse/#no-external-messages-mode>`__)
+"""
+
+from irctest import cases
+from irctest.numerics import ERR_CANNOTSENDTOCHAN
+
+
+class NoOutsideTestCase(cases.BaseServerTestCase):
+    @cases.mark_specifications("RFC1459")
+    def testNoOutsideMode(self):
+        # test the +n channel mode
+        self.connectClient("chanop", name="chanop")
+        self.joinChannel("chanop", "#chan")
+        self.sendLine("chanop", "MODE #chan +n")
+        self.getMessages("chanop")
+
+        self.connectClient("baz", name="baz")
+        # this message should be suppressed completely by +n
+        self.sendLine("baz", "PRIVMSG #chan :hi from baz")
+        replies = self.getMessages("baz")
+        reply_cmds = {reply.command for reply in replies}
+        self.assertIn(ERR_CANNOTSENDTOCHAN, reply_cmds)
```

Comment:
I'd be surprised if the CI passes for this one

Reply:
oh well, I guess it does pass for once.

Reply:
This is largely copied and pasted from the test for `+m`.